app/templates/results.html

```diff
@@ -44,8 +44,14 @@
             }).on('input', function (event) {
                 var canvas = document.querySelector('#' + event.target.dataset.canvas);
                 var ctx = canvas.getContext('2d');
-                ctx.clearRect(0, 0, canvas.width, canvas.height);
-                ctx.fillText(event.target.value, 10, 50);
+                var img = new Image();
```

Comment:
How does this fix that?

Reply:
The modification in the code introduces an `Image` object and sets up an `onload` event handler for it. This change is crucial for ensuring that the image is loaded into the canvas before any text is drawn. Here's a breakdown of how this addresses the issue:

1. **Image Loading**: By creating an `Image` object and assigning the source (`src`) from the corresponding image element, we ensure that the image data is fully loaded into the script's context.
2. **Onload Event**: The `onload` event handler is triggered once the image is completely loaded. Inside this handler:
   - **Draw Image**: The image is drawn onto the canvas using `ctx.drawImage(img, 0, 0, canvas.width, canvas.height)`. This ensures that the image covers the entire canvas, which resolves the issue of the gray background seen in the downloaded image.
   - **Text Overlay**: After the image is drawn, the text is then rendered on top of it at a specific position (`ctx.fillText(event.target.value, 10, 50)`). This placement can be adjusted if needed, but it starts the text overlay from the top-left of the canvas (slightly indented).

This approach ensures that every time the text input changes, the canvas is updated with the image first and then the text, maintaining the correct layering order and visibility in the downloaded image. This dynamic update is crucial for ensuring that the text appears in the correct position relative to the image, and it addresses the problem of static text positioning.